xrally_kubernetes/tasks/scenarios/pods.py

```diff
@@ -0,0 +1,118 @@
+            return time.mktime(dt)
+
+        if not conditions:
+            return []
+
+        start, finish = None, None
+        stat_dict = {
+            "kubernetes.initialized_pod": {},
+            "kubernetes.scheduled_pod": {},
+            "kubernetes.created_pod": {}
+        }
+        for d in conditions:
+            if d.type == "Initialized":
+                start = init_time = to_time(d.last_transition_time.timetuple())
+                stat_dict["kubernetes.initialized_pod"].update({
+                    "started_at": init_time
+                })
+            elif d.type == "PodScheduled":
+                scheduled = to_time(d.last_transition_time.timetuple())
+                stat_dict["kubernetes.scheduled_pod"].update({
+                    "started_at": scheduled
+                })
+                stat_dict["kubernetes.initialized_pod"].update({
+                    "finished_at": scheduled
+                })
+            elif d.type == "Ready":
+                finish = to_time(d.last_transition_time.timetuple())
+                stat_dict["kubernetes.created_pod"].update({
+                    "started_at": start,
+                    "finished_at": finish
+                })
+                stat_dict["kubernetes.scheduled_pod"].update({
+                    "finished_at": finish
+                })
+        return [{"name": k,
+                 "started_at": v["started_at"],
+                 "finished_at": v["finished_at"]}
+                for k, v in stat_dict.items()]
+
+    def _make_data_from_conditions(self, conditions):
+        """Make plot data from conditions made by parse method."""
+        data = [[] for _ in range(3)]
+        state_map = {
+            "kubernetes.initialized_pod": 0,
+            "kubernetes.scheduled_pod": 1,
+            "kubernetes.created_pod": 2
+        }
+
+        for e in conditions:
+            duration = e["finished_at"] - e["started_at"]
+            data[state_map[e["name"]]] = [e["name"], duration]
+        return data
+
+    def run(self, image, name=None, command=None, status_wait=True):
+        """Create pod, wait until it won't be running and then delete it.
+
+        :param image: pod's image
+        :param name: pod's custom name
+        :param command: array of strings, pod's command. Could be None iff
+               image has entrypoint
+        :param status_wait: wait pod status after creation
+        """
+        namespace = self.choose_namespace()
+
+        name = self.client.create_pod(
+            name,
+            namespace=namespace,
+            image=image,
+            command=command,
+            status_wait=status_wait
+        )
+
+        pod = self.client.get_pod(name, namespace=namespace)
+        conditions = self._parse_pod_status_conditions(pod.status.conditions)
+
+        self.client.delete_pod(
+            name,
+            namespace=namespace,
+            status_wait=status_wait
+        )
+
+        self.add_output(
```

Comment:
why is this called after delete pod?

Reply:
no sense where is called, but I can move it upwards

Reply:
In case of failure in the delete request, this data will be lost if it is called in the end.

Reply:
already fixed